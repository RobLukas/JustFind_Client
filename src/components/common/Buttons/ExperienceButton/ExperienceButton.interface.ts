interface ExperienceButtonProps {
  handleClick(): void;
}

export default ExperienceButtonProps;
