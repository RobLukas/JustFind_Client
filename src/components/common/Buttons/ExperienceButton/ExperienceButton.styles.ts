import styled from 'styled-components';

import SvgIcon from '@material-ui/icons/ExpandMore';

const ExperienceButton = styled.div`
  display: inline-block;
  position: relative;
`;

const DropDownButton = styled.button`
  color: ${({ theme }) => theme.styles.filter.button.color};
  background: ${({ theme }) => theme.styles.secondary.background};
  margin-left: 0;
  cursor: pointer;
  padding: 0 7px 0 12px;
  margin: 0;
  box-shadow: none;
  text-transform: none;
  height: 35px;
  align-items: center;
  font-size: 0.8rem;
  border-radius: 20px;
  display: flex;
  text-decoration: none;
  text-align: center;
  letter-spacing: 0.5px;
  outline: 0;
  border: 0;
  -webkit-tap-highlight-color: transparent;
  &.MuiSvgIcon-root {
    fill: ${({ theme }) => theme.styles.filter.button.color};
  }
  &:hover {
    background-color: rgba(255, 255, 255, 0.08);
    box-shadow: 0 2px 2px 0 rgba(0, 0, 0, 0.14), 0 1px 5px 0 rgba(0, 0, 0, 0.12),
      0 3px 1px -2px rgba(0, 0, 0, 0.2);
  }
`;

const Title = styled.span`
  vertical-align: super;
  margin: 0 3px;
`;

const MuiIcon = styled(SvgIcon)`
  &.MuiSvgIcon-root {
    font-size: 1.2rem;
    line-height: inherit;
  }
` as typeof SvgIcon;

export default { ExperienceButton, DropDownButton, Title, MuiIcon };
