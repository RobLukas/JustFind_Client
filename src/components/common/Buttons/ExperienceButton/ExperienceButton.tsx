import React, { FC } from 'react';
import TrendingUpIcon from '@material-ui/icons/TrendingUp';
import ExpandMoreIcon from '@material-ui/icons/ExpandMore';

import S from './ExperienceButton.styles';
import ExperienceButtonProps from './ExperienceButton.interface';

const ExperienceButton: FC<ExperienceButtonProps> = ({
  handleClick,
}: ExperienceButtonProps) => {
  return (
    <S.ExperienceButton>
      <S.DropDownButton onClick={handleClick}>
        <S.MuiIcon component={TrendingUpIcon} />
        <S.Title>{'Exp. level'}</S.Title>
        <S.MuiIcon component={ExpandMoreIcon} />
      </S.DropDownButton>
    </S.ExperienceButton>
  );
};

export default ExperienceButton;
