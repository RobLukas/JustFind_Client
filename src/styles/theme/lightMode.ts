export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
