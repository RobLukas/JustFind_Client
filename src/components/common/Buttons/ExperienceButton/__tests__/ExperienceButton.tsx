import React from 'react';
import 'jest-styled-components';
import { cleanup, render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import lightMode from 'styles/theme/lightMode';
import ExperienceButton from '../ExperienceButton';

describe('<ExperienceButton />', () => {
  afterEach(cleanup);
  it('render correctly', () => {
    const handleClick = jest.fn();

    const { container } = render(
      <ThemeProvider theme={lightMode}>
        <ExperienceButton handleClick={handleClick} />
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
