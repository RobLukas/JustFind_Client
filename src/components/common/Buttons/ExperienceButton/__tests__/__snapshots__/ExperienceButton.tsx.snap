// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ExperienceButton /> render correctly 1`] = `
.c0 {
  display: inline-block;
  position: relative;
}

.c1 {
  color: #777;
  background: #fff;
  margin-left: 0;
  cursor: pointer;
  padding: 0 7px 0 12px;
  margin: 0;
  box-shadow: none;
  text-transform: none;
  height: 35px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  font-size: 0.8rem;
  border-radius: 20px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-text-decoration: none;
  text-decoration: none;
  text-align: center;
  -webkit-letter-spacing: 0.5px;
  -moz-letter-spacing: 0.5px;
  -ms-letter-spacing: 0.5px;
  letter-spacing: 0.5px;
  outline: 0;
  border: 0;
  -webkit-tap-highlight-color: transparent;
}

.c1.MuiSvgIcon-root {
  fill: #777;
}

.c1:hover {
  background-color: rgba(255,255,255,0.08);
  box-shadow: 0 2px 2px 0 rgba(0,0,0,0.14),0 1px 5px 0 rgba(0,0,0,0.12), 0 3px 1px -2px rgba(0,0,0,0.2);
}

.c3 {
  vertical-align: super;
  margin: 0 3px;
}

.c2.MuiSvgIcon-root {
  font-size: 1.2rem;
  line-height: inherit;
}

<div
  class="c0"
>
  <button
    class="c1"
  >
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-root c2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16 6l2.29 2.29-4.88 4.88-4-4L2 16.59 3.41 18l6-6 4 4 6.3-6.29L22 12V6z"
      />
    </svg>
    <span
      class="c3"
    >
      Exp. level
    </span>
    <svg
      aria-hidden="true"
      class="MuiSvgIcon-root MuiSvgIcon-root c2"
      focusable="false"
      viewBox="0 0 24 24"
    >
      <path
        d="M16.59 8.59L12 13.17 7.41 8.59 6 10l6 6 6-6z"
      />
    </svg>
  </button>
</div>
`;
